pkg/controller/plan/validation.go

```diff
@@ -143,6 +147,39 @@ func (r *Reconciler) validate(plan *api.Plan) error {
 		return err
 	}
 
+	// Validate version only if migration is OCP to OCP
+	err = r.validateOCPVersion(plan)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) validateOCPVersion(plan *api.Plan) error {
+	if plan.IsSourceProviderOCP() && plan.Provider.Destination.Type() == api.OpenShift {
+		unsupportedVersion := libcnd.Condition{
+			Type:     unsupportedVersion,
+			Status:   True,
+			Reason:   NotSupported,
+			Category: Critical,
+			Message:  "Source version is not supported.",
+			Items:    []string{},
+		}
+
+		restCfg := ocp.RestCfg(plan.Referenced.Provider.Source, plan.Referenced.Secret)
+		clientset, err := kubernetes.NewForConfig(restCfg)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+
+		err = r.checkOCPVersion(clientset)
+		if err != nil {
+			plan.Status.SetCondition(unsupportedVersion)
+			return err
```

Comment:
the error can be cause because multiple error even before getting the version. i guess it's a matter of taste of you wish to differentiate between those cases.

Reply:
it's true but then it's a failure to validate rather than a validation failure